src/v/cloud_storage/remote_partition.cc

```diff
@@ -1086,14 +1101,23 @@ ss::future<storage::translating_reader> remote_partition::make_reader(
     std::ignore = deadline;
     vlog(
       _ctxlog.debug,
-      "remote partition make_reader invoked, config: {}, num segments {}",
+      "remote partition make_reader invoked (waiting for units), config: {}, "
+      "num segments {}",
       config,
```

Comment:
This is super helpful, but I'm also wary of how verbose it may be. At least looking at log_reader_config, there's a bunch of fields that we probably don't need logged every time if the main goal is a unique identifier.

Maybe we need some `short_string()` or something that just prints the offset and address or something

Reply:
Yes the config prints a lot of fields which is usually not of much use; sometimes their value is unexpectedly helpful though, for example recently  when working on a CI failure, the logs showed that the abort source had not been triggered before a certain point and was triggered after (via the `aborted` field). 

I think we definitely should prune the amount of info printed but also make sure we keep stuff which might be useful. I can follow up on this in a future PR if that seems acceptable.

Reply:
my contrib: we could move to fmt::fomatter and a format specifier {:s} for short representation, with the default begin the current long representation.

it's a bit of space trimming, bools are printed as T or F, some names are shorted.  


```cpp

template<>
struct fmt::formatter<kafka::fetch_config> {
    char presentation = 'u'; // 'u' for unchanged, 's' for short
    constexpr auto parse(format_parse_context& ctx) {
        auto it = ctx.begin();
        auto end = ctx.end();
        if (it != end && (*it == 's' || *it == 'u')) {
            presentation = *it++;
        }
        if (it != end && *it != '}') {
            throw format_error("invalid format");
        }
        return it;
    }

    constexpr static auto bool_s(bool val) { return val ? "T" : "F"; }
    auto format(kafka::fetch_config const& cfg, auto& ctx) const {
        if (presentation == 'u') {
            return fmt::format_to(
              ctx,
              R"({{"start_offset": {}, "max_offset": {}, "isolation_lvl": {}, "max_bytes": {}, "strict_max_bytes": {}, "skip_read": {}, "current_leader_epoch:" {}, "follower_read:" {}, "consumer_rack_id": {}, "abortable": {}, "aborted": {}, "client_address": {}}})",
              cfg.start_offset,
              cfg.max_offset,
              cfg.isolation_level,
              cfg.max_bytes,
              cfg.strict_max_bytes,
              cfg.skip_read,
              cfg.current_leader_epoch,
              cfg.read_from_follower,
              cfg.consumer_rack_id,
              cfg.abort_source.has_value(),
              cfg.abort_source.has_value()
                ? cfg.abort_source.value().get().abort_requested()
                : false,
              cfg.client_address.value_or("unknown"));
        }
        return fmt::format_to(
          ctx.out(),
          R"({{start_offset|max_offset:{}|{} isolation:{} max_bytes|strict:{}|{} skip_read:{} current_leader_epoch:{} follower_read:{} consumer_rack_id:{} abortable|aborted:{}|{} client_address:"{}"}})",
          cfg.start_offset,
          cfg.max_offset,
          cfg.isolation_level,
          cfg.max_bytes,
          bool_s(cfg.strict_max_bytes),
          bool_s(cfg.skip_read),
          cfg.current_leader_epoch,
          bool_s(cfg.read_from_follower),
          cfg.consumer_rack_id,
          bool_s(cfg.abort_source.has_value()),
          bool_s(
            cfg.abort_source.has_value()
              ? cfg.abort_source.value().get().abort_requested()
              : false),
          cfg.client_address.value_or("n/a"));
    }
};
```

Reply:
This looks good to me, I think this can be added in a follow up PR, wdyt? For example we might want to just skip the `client_address` if it is not set.

Reply:
yes of course